packages/react-theme-provider/src/components/ThemeProvider/useThemeProvider.ts

```diff
@@ -37,18 +38,8 @@ export const useThemeProvider = (
   const localTheme = state.theme;
 
   state.theme = mergeThemes(parentTheme, localTheme);
-  state.style = React.useMemo(() => {
-    // TODO: should we consider insertion to head?
-    //       - how to modify, remove styles?
-    //       - SSR rendering
-
-    // TODO: what variables should be rendered? Merged or only changed?
-    // TODO: how we will proceed with Portals?
-    return {
-      ...state.style,
-      ...themeToCSSVariables(state.theme),
-    };
-  }, [state.style, state.theme]);
+  const themeClassName = useThemeStyleTag(state.theme, state.document);
 
+  state.className = themeClassName;
```

Comment:
```suggestion
  state.themeClassName = themeClassName;
```

As `state.className` equals to `props.className`, we should use a different key on state.

Reply:
except in this case we want the className to be applied to the root slot of the `ThemeProvider` ? 

I don't really see why we should allow classNames to be passed from props

Reply:
> I don't really see why we should allow classNames to be passed from props

As this component renders DOM i.e. `slots.root` we should allow user's input (other DOM props will be passed to a slot).